cmd/grafana-app-sdk/project.go

```diff
@@ -411,53 +397,53 @@ func projectAddComponent(cmd *cobra.Command, args []string) error {
 	}
 
 	// Kind format
-	format, err := cmd.Flags().GetString("format")
-	if err != nil {
-		return err
-	}
-
-	// Plugin ID (optional depending on component)
-	pluginID, err := cmd.Flags().GetString("plugin-id")
+	format, err := cmd.Flags().GetString(formatFlag)
 	if err != nil {
 		return err
 	}
-	if len(pluginID) > 0 && !validNameRegex.MatchString(pluginID) {
-		fmt.Printf("plugin-id '%s' is not valid. Name must begin and end with an alphanumeric character, "+
-			"and only contain alphanumeric characters and _ or -", pluginID)
-		os.Exit(1)
-	}
 
-	kindGrouping, err := cmd.Flags().GetString("kindgrouping")
+	kindGrouping, err := cmd.Flags().GetString("grouping")
 	if err != nil {
 		return err
 	}
 	if kindGrouping != kindGroupingGroup && kindGrouping != kindGroupingKind {
-		return fmt.Errorf("--kindgrouping must be one of 'group'|'kind'")
+		return fmt.Errorf("--grouping must be one of 'group'|'kind'")
 	}
 
 	// Create the generator (used for generating non-static code)
 	var generator any
+	var manifestParser codegen.Parser[codegen.AppManifest]
 	switch format {
 	case FormatCUE:
 		parser, err := cuekind.NewParser()
 		if err != nil {
 			return err
 		}
-		generator, err = codegen.NewGenerator[codegen.Kind](parser.KindParser(true), os.DirFS(cuePath))
+		generator, err = codegen.NewGenerator[codegen.Kind](parser.KindParser(true), os.DirFS(sourcePath))
 		if err != nil {
 			return err
 		}
+		manifestParser = parser.ManifestParser()
 	default:
 		return fmt.Errorf("unknown kind format '%s'", format)
 	}
 
+	manifests, err := manifestParser.Parse(os.DirFS(sourcePath), selector)
+	if err != nil {
+		return fmt.Errorf("error parsing manifest '%s': %v", sourcePath, err)
+	}
+	if len(manifests) == 0 {
+		return fmt.Errorf("no manifest found in '%s'", sourcePath)
+	}
+	manifest := manifests[0]
```

Comment:
Any chance we get `len(manifests) > 1`?

Reply:
There's not a normal situation where this should happen (we only allow one selector, and the parser should only return one), if we do, taking the first one should be fine, unless we'd rather error.

Reply:
I think we could error (for future-proofing, since explicit failure is easier to debug than weird behaviour) but I don't think it's a blocker – feel free to leave it as-is.